react/src/components/ComputeSessionNodeItems/SessionIdleChecks.tsx

```diff
@@ -0,0 +1,185 @@
+import { useTranslation } from 'react-i18next';
+import { useFragment } from 'react-relay';
+
+type BaseExtra = any | null;
+type UtilizationExtra = {
+  resources: {
+    cpu_util: number[];
+    mem: number[];
+    cuda_util: number[];
+    cuda_mem: number[];
+    ipu_util: number[];
+    ipu_mem: number[];
+  };
+  thresholds_check_operator: 'and' | 'or';
+};
+export type IdleCheckItem = {
+  extra: BaseExtra | UtilizationExtra;
+  remaining: number | null;
+  remaining_time_type: 'expire_after' | 'grace_period';
+};
+export type IdleChecks = {
+  network_timeout?: IdleCheckItem;
+  session_lifetime?: IdleCheckItem;
+  utilization?: IdleCheckItem;
+};
+
+interface SessionIdleChecksProps {
+  sessionNodeFrgmt?: SessionIdleChecksNodeFragment$key | null;
+  sessionFrgmt?: SessionIdleChecksFragment$key | null;
+  direction?: 'row' | 'column';
+}
+
+const SessionIdleChecks: React.FC<SessionIdleChecksProps> = ({
+  sessionNodeFrgmt = null,
+  sessionFrgmt = null,
+  direction = 'row',
+}) => {
+  const { t } = useTranslation();
+  const { token } = theme.useToken();
+  const { mergedResourceSlots } = useResourceSlotsDetails();
+
+  const sessionNode = useFragment(
+    graphql`
+      fragment SessionIdleChecksNodeFragment on ComputeSessionNode {
+        id
+        idle_checks @since(version: "24.12.0")
+      }
+    `,
+    sessionNodeFrgmt,
+  );
+  const session = useFragment(
+    graphql`
+      fragment SessionIdleChecksFragment on ComputeSession {
+        id
+        idle_checks @since(version: "24.09.0")
+      }
+    `,
+    sessionFrgmt,
+  );
+
+  const idleChecks: IdleChecks = JSON.parse(
+    sessionNode?.idle_checks || session?.idle_checks || '{}',
+  );
+
+  const getIdleCheckTitle = (key: keyof IdleChecks) => {
+    if (key === 'network_timeout') return t('session.NetworkIdleTimeout');
+    else if (key === 'session_lifetime') return t('session.MaxSessionLifetime');
+    else return t('session.UtilizationIdleTimeout');
+  };
+
+  const getRemainingTimeTypeLabel = (Type: 'expire_after' | 'grace_period') => {
+    if (Type === 'expire_after') {
+      return t('session.ExpiresAfter');
+    } else {
+      return t('session.GracePeriod');
+    }
+  };
+
+  return (
+    <Flex direction={direction} align="stretch" gap="sm">
+      {_.map(idleChecks, (value: IdleCheckItem, key: keyof IdleChecks) => {
+        if (!value.remaining) return null;
+
+        return (
+          <Flex
+            key={key}
+            style={{ flex: 1 }}
+            direction="column"
+            align="stretch"
+          >
+            <Flex gap={'xxs'}>
+              <Typography.Text>{getIdleCheckTitle(key)}</Typography.Text>
+              {key === 'utilization' && (
+                <Tooltip
+                  title={
+                    <>
+                      {`${t('session.Utilization')} / ${t('session.Threshold')} (%)`}
+                      <br />
+                      {_.map(value.extra.resources, (resource, key) => {
```

Comment:
After creating the session, I got the `extra` value as null. Could you please check the type and add conditions if needed?
```
{
    "remaining": 3589.230596,
    "remaining_time_type": "expire_after",
    "extra": null
}
```

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/XqC2uNFuj0wg8I60sMUh/2b79d995-f18a-49a7-b752-483f98894f05.png)

Reply:
extra has a non-null value only when it is the result of a utilization check. This is caused by the idle checker returning null until the utilization calculation is complete.

Removed any from the BaseExtra type, so that it only has the type UtilizationExtra | null.